src/server/server_impl.go

```diff
@@ -47,20 +54,33 @@ func (server *server) Start() {
 	go func() {
 		addr := fmt.Sprintf(":%d", server.debugPort)
 		logger.Warnf("Listening for debug on '%s'", addr)
-		logger.Info(http.ListenAndServe(addr, server.debugListener.debugMux))
+		var err error
+		server.debugListener.listener, err = reuseport.Listen("tcp", addr)
+
+		if err != nil {
+			logger.Errorf("Failed to open debug HTTP listener: '%+v'", err)
+			return
+		}
+		logger.Info(http.Serve(server.debugListener.listener, server.debugListener.debugMux))
```

Comment:
perhaps grab the error in `err` and log it with `failed to start debug server: err` or something similar?